storage/mroonga/ha_mroonga.cpp

```diff
@@ -296,6 +297,7 @@ static PSI_mutex_info mrn_mutexes[] =
 #endif
 
 /* global variables */
+bool mrn_initialized = false;
```

Comment:
Continuing the comment in [3307](https://github.com/MariaDB/server/pull/3307):
> > Is this safe for multi-threading or CPU optimization? Is there a need to add mrn_mutexes similar to other global variables?
> > 
> Yes. It's safe for multi-threading. This is changed only by st_mysql_plugin::init and st_mysql_plugin::deinit and they aren't called with multiple threads at once.
> 
> I'm not sure for CPU optimization. What situation do you think about?

I mean you probably need to ensure the [memory barrier](https://en.wikipedia.org/wiki/Memory_barrier) using mutex or atomic operations to maintain ordering constraints and avoid inconsistent memory views between threads due to CPU and compiler optimizations?

Reply:
There's a `LOCK_plugin` around the initialization and de-initialization of all plugins and UDFs so there isn't actually a race condition here.

I'm still preferring the `plugin_is_ready` enhancement to the server below to facilitate consistent checking of relations between plugins.